Orphan and adopt pods with problematic selectors

tfussell opened the pull request:
Fixes https://github.com/giantswarm/giantswarm/issues/11154#issuecomment-645578936

## Checklist

- [x] Update changelog in CHANGELOG.md.

kopiczko commented on files/conf/k8s-addons:
```diff
@@ -70,6 +70,18 @@ do
+# upgraded if needed in a graceful manner.
+if [[ $(kubectl get ds -n kube-system kube-proxy -o jsonpath='{.spec.selector.matchLabels}') == "map[app.kubernetes.io/name:kube-proxy]" ]]; then
+  echo "Orphaning kube-proxy pods to be inherited by new deployment"
+  kubectl delete ds -n kube-system kube-proxy --cascade=false
```
How is it supposed to work. Are the labels for the new selector already present on the pods?

tfussell replied:
Yes, here was the PR https://github.com/giantswarm/k8scloudconfig/pull/695/files You can see that the new label was added but the old label, `k8s-app`, wasn't removed.

tfussell commented:
I just updated this branch and fixed some broken links in the CHANGELOG. The code is unchanged. I just confirmed that aws-operator and azure-operator didn't have any releases with the problematic selector label `app.kubernetes.io/name` which means this should only affect KVM. I have tested upgrades on KVM so I believe this is safe to merge and release.
